lib/feature/auth/sign_up/screen/sign_up_screen.dart

```diff
@@ -0,0 +1,364 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:split/core/widgets/base_stateful_screen_widget.dart';
+import 'package:split/feature/auth/auth_base.dart';
+import 'package:split/feature/auth/otp_verification/screen/otp_screen.dart';
+import 'package:split/feature/auth/sign_in/screen/sign_in_screen.dart';
+import 'package:split/feature/auth/sign_up/bloc/sign_up_bloc.dart';
+import 'package:split/feature/auth/widgets/app_elevated_button.dart';
+import 'package:split/feature/auth/widgets/app_text_form_field.dart';
+import 'package:split/feature/auth/widgets/social_button.dart';
+import 'package:split/res/app_asset_paths.dart';
+import 'package:split/res/app_colors.dart';
+import 'package:split/utils/locale/app_localization_keys.dart';
+import 'package:split/utils/validations/auth_validate.dart';
+
+class SignUpScreen extends StatelessWidget {
+  const SignUpScreen({super.key});
+  static const routeName = 'signUpScreen';
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) => SignUpBloc(),
+      child: const SignUpScreenWithBloc(),
+    );
+  }
+}
+
+class SignUpScreenWithBloc extends BaseStatefulScreenWidget {
+  const SignUpScreenWithBloc({super.key});
+
+  @override
+  BaseScreenState<SignUpScreenWithBloc> baseScreenCreateState() =>
+      _SignUpScreenWithBlocState();
+}
+
+class _SignUpScreenWithBlocState extends BaseScreenState<SignUpScreenWithBloc>
+    with AuthValidate {
+  final GlobalKey<FormState> formKey = GlobalKey();
+  AutovalidateMode autovalidateMode = AutovalidateMode.disabled;
+  String? email, password, name, phone;
+  @override
+  Widget baseScreenBuild(BuildContext context) {
+    return BlocConsumer<SignUpBloc, SignUpState>(
+      listener: (context, state) {
+        if (state is SignUpSuccessState) {
+          _openOtpScreen(context);
+        } else if (state is SignInScreenState) {
+          _openSignInScreen(context);
+        } else if (state is ValidateRegisterState) {
+          _register(context);
+        } else if (state is NotValidateRegisterState) {
+          autovalidateMode = AutovalidateMode.always;
+        }
+      },
+      buildWhen: (previous, current) {
+        if (current is NotValidateRegisterState ||
+            current is CheckAgreeState ||
+            current is NotCheckAgreeState) {
+          return true;
+        } else {
+          return false;
+        }
+      },
+      builder: (context, state) {
+        return AuthBase(
+          body: Container(
+            decoration: const BoxDecoration(
```

Comment:
pls, explain why didn't you add this code into the AuthBase because i've seen it alot!?
the container and its decoration